polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -168,13 +180,44 @@ pub fn worker_entrypoint(
 					Arc::clone(&condvar),
 					WaitOutcome::TimedOut,
 				)?;
+
+				let fd = stream.as_raw_fd();
+				let crash_event = unsafe {
+					crash_handler::make_crash_event(move |context| {
+						ALLOC.end_tracking();
+						if context.siginfo.ssi_signo as i32 == libc::SIGABRT {
+							// We're inside `SIGABRT` handler. The process state is compromised so
+							// no sudden movements please. Avoid allocations and make no assumptions
+							// based on the state of the stack.
+
+							// Send pre-encoded `Err(PrepareError::OutOfMemory)` to the host, close
+							// the connection, bail out.
+							libc::write(
+								fd,
+								b"\x02\x00\x00\x00\x00\x00\x00\x00\x01\x08" as *const _
+									as *const libc::c_void,
+								10,
+							);
+							libc::close(fd);
+							std::process::exit(1);
+						}
+						CrashEventResult::Handled(true)
+					})
+				};
+
+				let _crash_handler = CrashHandler::attach(crash_event).map_err(|_| {
+					io::Error::new(io::ErrorKind::Other, "Failed to install crash handler")
+				})?;
+
+				ALLOC.start_tracking(executor_params.prechecking_max_memory().map(|v| v as isize));
```

Comment:
Not sure things are worth complicating here... What is the failure scenario? The governance puts on chain a `PrecheckingMaxMemory` value that doesn't fit into `isize`? Well, as I like to say, if the network wants to commit suicide, it will find a way.
If we could escape with `::from()`, it would be okay, but `isize` obviously doesn't implement `From<u64>`.

Reply:
Well, mistakes do happen and it's good to practice defensive coding. On failure we could log a warning and return 0. I'd prefer that to potentially silently accepting a garbage value. To avoid clutter here, it could be extracted into a small helper. But, feel free to ignore - it's just a nit after all. I'm not very dogmatic, but it's good to stay aware of the better alternatives to `as` (`from` and `try_from`).